src/main.rs

```diff
@@ -70,19 +73,26 @@ fn run_cli() -> Result<(), String> {
     Command::Run { file, params } => {
       let code = load_file_code(&hvm(&file))?;
 
-      run_code(&code, false, params)?;
+      run_code(&code, false, params, parse_gb(&cli_matches.memory))?;
       Ok(())
     }
 
     Command::Debug { file, params } => {
       let code = load_file_code(&hvm(&file))?;
 
-      run_code(&code, true, params)?;
+      run_code(&code, true, params, parse_gb(&cli_matches.memory))?;
       Ok(())
     }
   }
 }
 
+fn parse_gb(raw: &str) -> usize {
```

Comment:
How hard would be for us to interpret units?

Reply:
If we are going to only support GB values we could drop the suffix altogether.

Also, `clap` seems to have support for custom parsing/validation (which would give better error messages).

Reply:
https://github.com/clap-rs/clap/blob/v3.1.15/examples/tutorial_derive/04_02_validate.rs

Reply:
> How hard would be for us to interpret units?

How many units do you think interesting?

Reply:
How many units do you think interesting? Particularly I think that MB and GB would be sufficient.

Reply:
I agreed that it would be sufficient. 
But I think if you just associate `K`, `M`, and `G` with the exponents (a list would suffice) it would be already more elegant and generic.
I think we should only use the prefix `M` instead of  `MB`, that also simplifies it.

Reply:
I think that would be good use the pattern of the International System of Units.

Reply:
The units should be [binary](https://en.wikipedia.org/wiki/Binary_prefix) actually.